### PR TITLE
Issue #11214: Specified violation message in GenericWhitespaceCheck

### PR DESCRIPTION
**Pull request by AmrDeveloper**
Issue #11214  Specified violation message in GenericWhitespaceCheck

### CODE REVIEW

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java [hunk 1]
```diff
@@ -63,10 +63,10 @@ public int getConstructor(Class<?>... parameterTypes)
     public static class IntEnumValueType<E extends Enum<E> & IntEnum> {
     }
 
-    public static class IntEnumValueType2<E extends Enum<E>& IntEnum> { // violation
+    public static class IntEnumValueType2<E extends Enum<E>& IntEnum> { // violation ''&' .*'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java:66]: message is too cryptic.
Lets think on line wrapping and chaging names in this line to let message be readable.

same for other lines.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java (hunk 1 above)
Comment (AmrDeveloper) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java:66]: Can i change them to

```java
public static class EnumVal<E extends Enum<E>& IntEnum> { // violation ''&' is not preceded .*'

public static class EnumVal2<E extends Enum<E>  & IntEnum> { // violation ''>' is followed .*'
 ```
 
 The line length will be 100 and 99

Comment (AmrDeveloper) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java:66]: In 

```
final Supplier<Optional<Integer>> function2 = Optional::<Integer> empty; // violation ''>' .*'
```

Can be 

```
final Supplier<Optional<Integer>> f2 = Optional::<Integer>  // violation ''>' is followed .*'
empty;
```

And len will be 98

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java:66]: try to remove `public static ` this Check does not care about them.

Comment (AmrDeveloper) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java:66]: > try to remove `public static ` this Check does not care about them. We need to remove or shorten all that does not matter for current Check/validation.

Great and the other we can shorten the name and remove final

```java
Supplier<Optional<Integer>> f2 = Optional::<Integer> empty; // violation ''>' is followed .*'
```

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java [hunk 2]
```diff
@@ -63,10 +63,10 @@ public int getConstructor(Class<?>... parameterTypes)
     public static class IntEnumValueType<E extends Enum<E> & IntEnum> {
     }
 
-    public static class IntEnumValueType2<E extends Enum<E>& IntEnum> { // violation
+    class IntEnumValueType2<E extends Enum<E>& IntEnum> { // violation ''&' is not preceded .*'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java:66]: IntEnumValueType2 -> IntEVT2
IntEnum -> IntE

please do all to give all space to message.

This check does not care about types names, even human who read - do not care.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java (hunk 2 above)
Comment (AmrDeveloper) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java [hunk 3]
```diff
@@ -84,13 +84,13 @@ public void beforeAndAfter() {
...
-    Object notOkStart = new<String>Object(); // violation
-    Object notOkEnd = new <String> Object(); // violation
+    Object notOkStart = new<String>Object(); // violation ''<' is not preceded with whitespace.'
+    Object notOkEnd = new <String> Object(); // violation ''>' is followed by whitespace.'
     Object notOkStartAndEnd = new<String> Object(); // 2 violations
     Object okWithPackage = new <String>java.lang.Object();
     Object ok2 = new <String>Outer.Inner();
-    Object notOkStart2 = new<String>Outer.Inner(); // violation
-    Object notOkEnd2 = new <String> Outer.Inner(); // violation
+    Object notOkStart2 = new<String>Outer.Inner(); // violation ''<' is not preceded .*'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java:92]: keep doing.

**Review by romani — CHANGES_REQUESTED** [review]
let make it simple:
all lines should have full message

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java (hunk 3 above)
Comment (AmrDeveloper) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceDefault.java]: > let make it simple: all lines should have full message

Done, now all of them have a full message

**Review by romani — APPROVED** [review]
Ok to merge if CI pass

**Comment by romani** [comment]
Wercker failing in git clone, we can skip this CI for this update

**Pull request merged**